api/src/services/email/index.ts

```diff
@@ -1,65 +1,132 @@
-import * as Postmark from 'postmark';
+import EmailDispatcher from './dispatcher';
 import { EmailTemplate, Templates } from './templates';
+import { Config } from '../../../config';
+import { User, CommunityBusiness } from '../../models';
+import { RoleEnum } from '../../models/types';
+import { AppEnum } from '../../types/internal';
 
-type Attachment = {
-  name: string,
-  content: string,
-  contentType: string,
-};
 
-type Email = {
-  from: string,
-  to: string,
-  templateId: EmailTemplate,
-  templateModel?: object,
-  attachments?: Attachment[],
-};
+export type EmailService = {
+  newVisitor:
+    (c: Config, v: User, a: User, cb: CommunityBusiness, at: string) => Promise<void>
 
-type EmailServiceConfig = {
-  apiKey: string,
-};
+  visitorReminder:
+    (c: Config, v: User, cb: CommunityBusiness, at: string) => Promise<void>
 
-export type EmailService = {
-  send: (emailOptions: Email) => Promise <Postmark.Models.MessageSendingResponse>,
-  sendBatch: (emailOptions: Email []) => Promise <Postmark.Models.MessageSendingResponse[]>,
-};
+  newVolunteer:
+    (c: Config, v: User, a: User, cb: CommunityBusiness) => Promise<void>
 
-type EmailInitialiser = {
-  init: (config: EmailServiceConfig) => EmailService,
-};
+  newCbAdmin:
+    (c: Config, a: User, cb: CommunityBusiness, t: string) => Promise<void>
 
-const emailKeyMap = (a: Email): Postmark.TemplatedMessage => ({
-  To: a.to,
-  From: a.from,
-  TemplateId: a.templateId,
-  TemplateModel: a.templateModel,
-  Attachments: (a.attachments || []).map(toPostmarkAttachment),
-});
+  addRole:
+    (c: Config, a: User, cb: CommunityBusiness, r: RoleEnum, t: string) => Promise<void>
 
-const toPostmarkAttachment = (a: Attachment): Postmark.Attachment => ({
-  Name: a.name,
-  ContentID: null,
-  Content: a.content,
-  ContentType: a.contentType,
-});
+  resetPassword:
+    (c: Config, ap: AppEnum, a: User, t: string) => Promise<void>
+};
 
-const emailInitialiser: EmailInitialiser = {
-  init: (config: EmailServiceConfig) => {
-    const emailClient = new Postmark.ServerClient(config.apiKey);
 
-    return {
-      send: async (emailOptions: Email) => {
-        const postmarkOptions = emailKeyMap(emailOptions);
+const Service: EmailService = {
```

Comment:
Why's this called `Service` rather than email service? If we add alternative services to our api this will get confusing.

Reply:
I called it that to disambiguate between the type and the actual object. The name is internal to the module only. When it's imported you can call it whatever you like.